### PR TITLE
CIVDEV-322 - Prepare Dev Environment

### PR DESCRIPTION
**Pull request by HilariousGil**
- Added an .env file ready with our desired values `.env.civity`
  - Added CKAN__SITE_URL=http://localhost:5000
- Changed base image for `ckan/ckan-base:2.9.9-dev` instead of 2.10
- Added file `ckan/setup/start_ckan_development.sh` to apply couple changes
  - Running on port 5000
  - Update Ckan Site URL
- Documentation on `Getting Started`
- Opened Ports for direct access to `SOLR` (8983) and `Redis`(6379)

The original `CKAN_SITE_URL` not being picked up might be some mistake of mine, we'll have a look at it, and see if it's worth creating an Issue + PR for the official Repository

Open for discussion is the version of SOLR:
- Currently using `SOLR_IMAGE_VERSION=2.10-solr9`

### CODE REVIEW

File: .env.civity [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+# Test database connections
+TEST_CKAN_SQLALCHEMY_URL=postgres://ckan:ckan@db/ckan_test
+TEST_CKAN_DATASTORE_WRITE_URL=postgresql://ckan:ckan@db/datastore_test
+TEST_CKAN_DATASTORE_READ_URL=postgresql://datastore_ro:datastore@db/datastore_test
+
+# CKAN core
+#CKAN_VERSION=2.10.0
+CKAN_SITE_ID=default
+CKAN_SITE_URL=http://localhost:5000
+CKAN__SITE_URL=http://localhost:5000
```
Comment (SaalTANK) [inline .env.civity:39]: This seems like to be the same 'variable', so why is this here twice?

File: .env.civity [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+CKAN_STORAGE_PATH=/var/lib/ckan
+CKAN_SMTP_SERVER=smtp.corporateict.domain:25
+CKAN_SMTP_STARTTLS=True
+CKAN_SMTP_USER=user
+CKAN_SMTP_PASSWORD=pass
+CKAN_SMTP_MAIL_FROM=ckan@localhost
+TZ=UTC
+
+# Solr
+SOLR_IMAGE_VERSION=2.9-solr9
```
Comment (SaalTANK) [inline .env.civity:58]: Wasn't there some discussion about which solr version to use?

File: README.md [hunk 3]
```diff
@@ -19,6 +20,34 @@
...
+	git clone https://github.com/CivityNL/ckan-docker.git
+    cd ckan-docker
+
+Checkout `civity` Branch
+
+    git checkout civity
+
+Create a copy of `.env.civity` file next to it and rename it to `.env`
+
+	cp .env.civity .env
```
Comment (SaalTANK) [inline README.md:38]: If we really want to have it quick, shouldn't we just rename this `.env.civity` to `.env`? In that case this step is obsolete

File: README.md [hunk 4]
```diff
@@ -19,6 +20,34 @@
...
+
+Clone the Repository and go to the root directory.
+
+	git clone https://github.com/CivityNL/ckan-docker.git
+    cd ckan-docker
+
+Checkout `civity` Branch
+
+    git checkout civity
+
```
Comment (SaalTANK) [inline README.md:35]: Also might be easier to make this `git clone --branch civity https://github.com/CivityNL/ckan-docker.git`?

File: ckan/Dockerfile.dev [hunk 5]
```diff
@@ -56,4 +56,6 @@ RUN for d in $APP_DIR/patches/*; do \
                 cd $SRC_DIR/`basename "$d"` && echo "$0: Applying patch $f to $SRC_DIR/`basename $d`"; patch -p1 < "$f" ; \
             done ; \
         fi ; \
-    done
\ No newline at end of file
+    done
+
+COPY setup/start_ckan_development.sh ${APP_DIR}
```
Comment (SaalTANK) [inline ckan/Dockerfile.dev]: If I read the original Dockerfile it feels like the intend was to install any extensions already earlier in this script (see: https://github.com/CivityNL/ckan-docker/blob/master/ckan/Dockerfile.dev#L15)

File: docker-compose.dev.yml [hunk 6]
```diff
@@ -65,12 +65,16 @@ services:
     volumes:
       - solr_data:/var/solr
     restart: unless-stopped
+    ports:
+      - "8983:8983"
```
Comment (SaalTANK) [inline docker-compose.dev.yml:69]: Why exposing the ports? Is this something specific we feel we need to add?

File: docker-compose.dev.yml [hunk 7]
```diff
@@ -65,12 +65,16 @@ services:
...
     healthcheck:
       test: ["CMD", "wget", "-qO", "/dev/null", "http://localhost:8983/solr/"]
 
   redis:
     container_name: ${REDIS_CONTAINER_NAME}
     image: redis:${REDIS_VERSION}
     restart: unless-stopped
+    ports:
+      - "6379:6379"
```
Comment (SaalTANK) [inline docker-compose.dev.yml]: Why exposing the ports? Is this something specific we feel we need to add?

**Review by SaalTANK — COMMENTED** [review]
Left a couple of comments, nothing breaking. Also have a lot of comments on the 'core' structure, but I've left that out for now

File: README.md (hunk 4 above)
Comment (HilariousGil) [reply README.md:35]: I will update this.

File: docker-compose.dev.yml (hunk 7 above)
Comment (HilariousGil) [reply docker-compose.dev.yml]: remoed.

File: README.md (hunk 3 above)
Comment (HilariousGil) [reply README.md:38]: keep in line with community .gitignore

File: .env.civity (hunk 1 above)
Comment (HilariousGil) [reply .env.civity:39]: Will remove one of them, and also create issue in Community Repo.
To see why it does not get picked up.

Comment (HilariousGil) [reply .env.civity:39]: Also test with ckanet-envvars to see if it can catch it.
